### PR TITLE
improvement: add tests for business logic (part 2)(SDKCF-1880)

### PR DESCRIPTION
**Pull request by imaginaris**
Enabled TestHost for UIView testing
changed display logic for accessibilityCompatibleDisplay flag - ~~To be tested~~ Tested.

### CODE REVIEW

File: RInAppMessaging/Classes/Views/BaseView.swift [hunk 1]
```diff
@@ -51,7 +52,14 @@ internal extension BaseView {
                 return
         }
 
-        let parentView = accessibilityCompatible ? (window.subviews.first ?? window) : window
+        var parentView: UIView = window
+
+        if accessibilityCompatible,
+            let mainSubview = window.subviews.first(
+                where: { !$0.isKind(of: NSClassFromString("UITransitionView")!) }) {
+            parentView = mainSubview
```
Comment (donnie-jp) [inline RInAppMessaging/Classes/Views/BaseView.swift]: The above logic should have an explanatory comment.

File: Tests/LockableTests.swift [hunk 2]
```diff
@@ -18,21 +18,35 @@ private class LockableTestObject: Lockable {
...
                 })
-                CommonUtility.lock(resourcesIn: [lockableObject]) {
-                    sleep(2)
-                    lockableObject.append(3)
-                }
+
+                lockableObject.resourcesToLock.forEach { $0.lock() }
+                sleep(2)
+                lockableObject.append(3)
+                lockableObject.resourcesToLock.forEach { $0.unlock() }
```
Comment (donnie-jp) [inline Tests/LockableTests.swift]: After it's unlocked it processes the stored operation (append 4)?

File: Tests/ViewTests.swift [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+                it("will save onDismiss action") {
+                    waitUntil { done in
+                        let onDismiss = { done() }
+                        view.show(accessibilityCompatible: false, onDismiss: onDismiss)
+                        expect(view.onDismiss).toNot(beNil())
+                        view.onDismiss?()
+                    }
+                }
+
+                it("will add itself to the UIWindow's view when `isUsingAutoLayout` is false") {
```
Comment (donnie-jp) [inline Tests/ViewTests.swift]: Not `accessibilityCompatible` is false?

File: Tests/ViewTests.swift [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+                    let window = UIApplication.shared.keyWindow
+                    view.show(accessibilityCompatible: false, onDismiss: {})
+                    expect(window?.subviews).to(containElementSatisfying({ $0 is BaseView }))
+                }
+
+                it("will add itself to the UIWindow's view subview when `isUsingAutoLayout` is true") {
```
Comment (donnie-jp) [inline Tests/ViewTests.swift]: Not `accessibilityCompatible` is true?

File: RInAppMessaging/Classes/Views/BaseView.swift (hunk 1 above)
Comment (imaginaris) [reply RInAppMessaging/Classes/Views/BaseView.swift]: Ok. Done

File: Tests/LockableTests.swift (hunk 2 above)
Comment (imaginaris) [reply Tests/LockableTests.swift]: Yes, the other thread with `append(4)` is waiting synchronously for the unlock() call

File: Tests/ViewTests.swift (hunk 3 above)
Comment (imaginaris) [reply Tests/ViewTests.swift]: Oh right! Wrong flag

**Pull request merged**